lib/private/Authentication/Token/DefaultTokenProvider.php

```
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @author Christoph Wurst <christoph@owncloud.com>
+ *
+ * @copyright Copyright (c) 2016, ownCloud, Inc.
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OC\Authentication\Token;
+
+use OCP\ILogger;
+
+class DefaultTokenProvider implements IProvider {
+
+	/** @var ILogger $logger */
+	private $logger;
+
+	public function __construct(ILogger $logger) {
+		$this->logger = $logger;
+	}
+
+	public function generateToken() {
+		return 'rq778zYGX8';
+	}
+
+	public function validateToken($token) {
+		$this->logger->debug('validating default token <' . $token . '>');
+		if ($token === 'rq778zYGX8') {
+			$this->logger->debug('token valid');
+			return 'user';
+		} else {
+			$this->logger->warning('invalid token');
```

Interesting. We might indeed want to look invalid token logins. For this we however need a mapping to avoid getting like a ton of warnings in case we have two providers. (such as the remember me token for example)

Also worth thinking about whether we want to throw some kind of "invalid auth" event at some point. But that's unrelated, just me thinking out loud :)